models/product_product.py

```diff
@@ -4,4 +4,18 @@
 class ProductProduct(models.Model):
     _inherit = 'product.product'
 
-    flower_id = fields.Many2one('flower')
\ No newline at end of file
+    def _get_gardeners_domain(self):
+        return [('id', 'in', self.env.ref('btco_sallys_flower_shop.gardner_user').users.ids)]
```

Comment:
This domain is confusing, why not use the field groups_id available on res.users directly? 

```suggestion
return [('groups_id', 'in', self.env.ref('btco_sallys_flower_shop.gardner_user').id)]
```